hmrc-mongo-test-play-28/src/main/scala/uk/gov/hmrc/mongo/test/MongoSupport.scala

```diff
@@ -104,3 +100,36 @@ trait IndexedMongoQueriesSupport extends MongoSupport with BeforeAndAfterAll {
     super.afterAll()
   }
 }
+
+trait TtlIndexedMongoSupport extends MongoSupport with TestSuite {
+  protected def collectionName: String
+
+  protected def checkTtlIndex: Boolean
+
+  override def withFixture(test: NoArgTest): Outcome =
+    super.withFixture(test) match {
+      case Succeeded if checkTtlIndex =>
+        (for {
+           was <- updateIndexPreference(false)
+           res <- MongoUtils.getTtlState(mongoComponent, collectionName, checkType = true)
+                    .map(res =>
```

Comment:
Is `Failed` a side effect?
```
             .foreach {
               case res if res.isEmpty => Failed(s"No ttl indices were found for collection $collectionName")
               case res if res.exists(k => k._2.isInstanceOf[TtlState.InvalidType]) =>
                 //collect all invalid and report all failures
                 Failed(s"ttl index for collection $collectionName points at $k which has type '$v', it should be 'date'")
               case res => res
             }
```

Would something like this be clearer?

Reply:
updated to report all invalid indexes together